compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -190,10 +197,8 @@ where
         let state = self.state;
         let job = {
             let mut shard = state.active.lock_shard_by_value(&self.key);
-            let job = match shard.remove(&self.key).unwrap() {
-                QueryResult::Started(job) => job,
-                QueryResult::Poisoned => panic!(),
-            };
+            let job = shard.remove(&self.key).unwrap().unwrap();
```

Comment:
This `.unwrap().unwrap()` is more awkard than the previous code, imo.

Reply:
Yeah it is weird